apps/web/public/sw.js

```diff
@@ -0,0 +1,411 @@
+/* Service Worker */
+
+const version = 1;
+const dbVersion = 2;
+const database = "inbox-zero";
+let DB = null;
+let dynamicName = `dynamicCache`;
+let IMAGE_CACHE = "IMAGE_CACHE";
+const LABEL_STORE = {
+  name: "labels",
+  key: "id",
+  indexes: [],
+};
+const EMAIL_STORE = {
+  name: "emails",
+  key: "gmailMessageId",
+  indexes: [{ name: "timestampIDX", property: "timestamp", params: {} }],
+};
+const ALL_STORES = [LABEL_STORE, EMAIL_STORE];
+
+const LABELS_UPDATED = "LABELS_UPDATED";
+const MAILS_UPDATED = "MAILS_UPDATED";
+
+const MESSAGES = { LABELS_UPDATED, MAILS_UPDATED };
+
+const TIME_CONSTANT = {
+  // in ms
+  hour: 3600_000,
+  day: 86400_000,
+  week: 604800_000,
+  month: 2629743_000,
+  year: 31556926_000,
+};
+
+self.addEventListener("install", (event) => {
+  console.log(`Version ${version} installed`);
+  self.skipWaiting();
+});
+
+self.addEventListener("activate", (event) => {
+  //service worker is activated
+  event.waitUntil(Promise.resolve().then(openDB));
+  console.log("activated");
+});
+
+self.addEventListener("fetch", (event) => {
+  //service worker intercepted a fetch call
+  console.log(event.request.url);
+  if (!isAuth(event.request.url))
```

Comment:
skips all the auth request and callback request coming from client and gmail server.